unstable/wlr-workspace-unstable-v1.xml

```diff
@@ -0,0 +1,226 @@
+        sent.
+
+        This allows changes to one or more zwlr_workspace_group_handle_v1 properties to
+        be seen as atomic, even if they happen via multiple events. In particular, an
+        output moving from one workspace group to another sends an output event and an
+        output_remove event to the two zwlr_workspace_group_handle_v1 objects in
+        question. The compositor sends both done events only after both other events.
+      </description>
+    </event>
+
+    <event name="destroyed">
+      <description summary="this workspace group has been destroyed">
+        This event means the zwlr_workspace_group_handle_v1 has been destroyed. It is
+        guaranteed there won't be any more events for this
+        zwlr_workspace_group_handle_v1. The zwlr_workspace_group_handle_v1 becomes inert
+        so any requests will be ignored except the destroy request.
+      </description>
+    </event>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the zwlr_workspace_group_handle_v1 object">
+        Destroys the zwlr_workspace_group_handle_v1 object.
+
+        This request should be called either when the client does not want to use the
+        workspace_group anymore or after the closed event to finalize the destruction of
+        the object.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="zwlr_workspace_handle_v1" version="1">
+    <description summary="a workspace belonging to a workspace group">
+      A zwlr_workspace_handle_v1 object represents a a workspace belonging to a
+      workspace group.
+
+      Each workspace group has a number, conveyed to the client with the number event,
+      and a list of states, conveyed to the client with the state event. The client may
+      request the compositor to activate the workspace.
+    </description>
+
+    <event name="number">
+      <description summary="workspace number changed">
+        This event is emitted immediately after the zwlr_workspace_handle_v1 is created
+        and whenever the number of the workspace changed. Workspaces within a workspace
+        group are uniquely and contiguously numbered starting from zero.
+      </description>
+      <arg name="number" type="int"/>
+    </event>
+
+    <event name="state">
+      <description summary="the state of the workspace changed">
+        This event is emitted immediately after the zwlr_workspace_handle_v1 is created
+        and each time the workspace state changes, either because of a compositor action
+        or because of a request in this protocol.
+      </description>
+      <arg name="state" type="array"/>
+    </event>
+
+    <enum name="state">
+      <description summary="types of states on the workspace">
+        The different states that a workspace can have.
+      </description>
+
+      <entry name="active" value="0" summary="the workspace is active"/>
```

Comment:
The overloading of "active" here is weird. It seems like "active" means some combination of "visible on its output" and "contains toplevels".

Reply:
Maybe this should be a bitfield?

Reply:
the state event gives an array of states, so they are not exclusive. I'm copying how toplevel states are conveyed in the foreign toplevel management protocol.

Reply:
I think the protocol just needs to clarify what is "active"

Reply:
Ok, I've defined a little better in the overall protocol description what 'active' means, hopefully that is satisfactory. Though really it is up to the compositor what it means, so the definition is not super precise.